conjure-undertow-processor/src/test/resources/com/palantir/conjure/java/undertow/processor/sample/DefaultDecoderServiceEndpoints.java.generated

```diff
@@ -246,6 +266,14 @@ public final class DefaultDecoderServiceEndpoints implements UndertowService {
                     "optionalStringParam", ParamDecoders.optionalStringCollectionParamDecoder(runtime.plainSerDe()));
             this.decoderParamDeserializer = new HeaderParamDeserializer<>(
                     "decoderParam", DefaultDecoderService.StringCollectionDecoder.INSTANCE);
+            this.floatHeaderBoxedDeserializer = new HeaderParamDeserializer<>(
+                    "floatHeaderBoxed",
+                    ParamDecoders.complexCollectionParamDecoder(runtime.plainSerDe(), Float::valueOf));
+            this.floatHeaderUnboxedDeserializer = new HeaderParamDeserializer<>(
+                    "floatHeaderUnboxed",
+                    ParamDecoders.complexCollectionParamDecoder(runtime.plainSerDe(), Float::parseFloat));
```

Comment:
`Float` and `float` use slightly separate codepaths, the former is handled using generic `valueOf` detection, while the latter is special cased.